spec/gtfs_parser_spec.rb

```diff
@@ -14,5 +13,31 @@
         expect(result[%w[31 31 0 Sunderland]]).is_a? Array
       end
     end
+    context 'with a trip ending at a stop and another beginning at the stop' do
+      let(:result) { find_departures(%w[stop_id]) }
+      it 'does not throw out the departure at the start of the trip' do
+        expect_any_instance_of(GtfsParser)
+          .to receive(:find_trips_operating_today).with('stop_id')
+          .and_return('trip_id1' => :route_data1, 'trip_id2' => :route_data2)
+        row0 = { 'trip_id' => 'trip_id1', 'stop_id' => 'something else' }
+        # row1 marks the end of a trip. Since the trip ends at the stop, it is
+        # not a departure and should not be included.
+        row1 = { 'trip_id' => 'trip_id1', 'stop_id' => 'stop_id',
+                 'departure_time' => '07:41:00'}
+        # row2 is the beginning of a trip, so a perfectly legitimate departure,
+        # and just happens to come after a trip end. It should be included.
+        row2 = { 'trip_id' => 'trip_id2', 'stop_id' => 'stop_id',
+                 'departure_time' => '08:21:00'}
+        row3 = { 'trip_id' => 'trip_id2', 'stop_id' => 'something else' }
+        iterator = double
```

Comment:
Why do you have to specify that `double` is `iterator` if you only use it twice? 

Is that so someone could go in later and change the iterator if more than one stop_id is supplied? Or if the data structure is changed?